email_processing.py

```diff
@@ -0,0 +1,197 @@
+    SENDER_EMAIL_PASSWORD = os.environ.get('SENDER_EMAIL_PASSWORD')
+
+    # -------------- END config
+
+    def get_login_data(self):
+        return {
+            'username': self.ROSSUM_USER,
+            'password': self.ROSSUM_PASSWORD,
+        }
+
+    def get_access_token(self):
+        user = self.ROSSUM_USER
+
+        login_api_url = f'https://{self.API_URL}/v1/auth/login'
+        login_data = self.get_login_data()
+
+        print('Performing authentication at RossumAPI ... ')
+        try:
+            r = requests.post(login_api_url, data=login_data)
+            # Get HTTP status code of a domain + path + data
+            if r.status_code not in range(200, 299):
+                raise Exception("Could not authenticate client.")
+                return False
+
+            response_from_elis = r.json()
+            token = response_from_elis['key']
+            self.access_token = token
+            print(f"User {user} has been successfully authenticated.")
+            return token
+        except requests.exceptions.RequestException:
+            raise Exception(f'Failed to connect to {login_api_url}') from None
+
+
+    def import_file_by_email(self):
+        SENDER_EMAIL_ADDRESS = self.SENDER_EMAIL_ADDRESS
+        SENDER_EMAIL_PASSWORD = self.SENDER_EMAIL_PASSWORD
+        EMAIL = self.EMAIL
+        FILE_NAME = self.FILE_NAME
+        msg = EmailMessage()
+        msg['Subject'] = 'Sample invoice file'
+        msg['From'] = SENDER_EMAIL_ADDRESS
+        msg['To'] = EMAIL
+
+        msg.set_content('Process data from the attached file')
+
+        with open(FILE_NAME, 'rb') as f:
+            file_data = f.read()
+            file_name = f.name
+            msg.add_attachment(file_data, maintype='application', subtype='octet-stream', filename=file_name)
+        try:
+            print(f"Sending email with an attachment {FILE_NAME} to {EMAIL}")
+            with smtplib.SMTP_SSL('smtp.gmail.com', 465) as smtp:
+                smtp.login(SENDER_EMAIL_ADDRESS, SENDER_EMAIL_PASSWORD)
+                smtp.send_message(msg)
+                print(f'Message has been succesfully delivered to.')
+
+        except smtplib.SMTPAuthenticationError:
+            print('Username and Password not accepted.')
+
+    def list_all_documents(self):
+        """ This function retreives all documents from requesting endpoint and writes it to list as class attribute.
+        """
+        access_token = self.access_token
+        all_files = self.all_files
+        documents_endpoint = f'https://{self.API_URL}/v1/documents'
+        headers = {
+            "Authorization": f"token {access_token}"
+        }
+        try:
+            r = requests.get(documents_endpoint, headers=headers)
+            if r.status_code in range(200, 299):
+                # Pagination count to check if there is more then 1 page otherwise file won't be found
+                pagination_count = r.json()['pagination']['total_pages']
```

Comment:
in order to limit nesting (and following the [zen of python ](https://www.python.org/dev/peps/pep-0020/#the-zen-of-python)
> Flat is better than nested.

)
one can use the so called [guard clause](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html) and leave the code behind the clause one level less nested.
```suggestion
            if not r.ok:
                return
            # Pagination count to check if there is more then 1 page otherwise file won't be found
            pagination_count = r.json()['pagination']['total_pages']
            ...
```